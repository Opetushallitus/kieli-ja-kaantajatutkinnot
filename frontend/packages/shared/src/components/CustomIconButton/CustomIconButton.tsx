import { QuestionMark } from '@mui/icons-material';
import { IconButton, IconButtonProps } from '@mui/material';
import { FC, forwardRef } from 'react';

const CustomIconButton: FC<IconButtonProps> = forwardRef(
  (props: IconButtonProps, ref) => {
    return (
      <IconButton ref={ref} {...props} aria-disabled={props.disabled}>
        {props.children ?? <QuestionMark />}
      </IconButton>
    );
  }
);

CustomIconButton.displayName = 'CustomIconButton';
export { CustomIconButton };
