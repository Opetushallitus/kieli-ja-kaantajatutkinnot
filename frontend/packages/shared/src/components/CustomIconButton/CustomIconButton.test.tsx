import HelpOutlineIcon from '@mui/icons-material/HelpOutline';
import renderer from 'react-test-renderer';

import { CustomIconButton } from './CustomIconButton';

describe('CustomIconButton', () => {
  it('should render correctly with QuestionMark icon by default', () => {
    const tree = renderer.create(<CustomIconButton />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render correctly with Icon as child', () => {
    const tree = renderer
      .create(
        <CustomIconButton>
          <HelpOutlineIcon />
        </CustomIconButton>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
