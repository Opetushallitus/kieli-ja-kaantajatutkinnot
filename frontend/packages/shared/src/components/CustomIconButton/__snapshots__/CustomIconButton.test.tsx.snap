// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomIconButton should render correctly with Icon as child 1`] = `
<button
  className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
  disabled={false}
  onBlur={[Function]}
  onContextMenu={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <svg
    aria-hidden={true}
    className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
    data-testid="HelpOutlineIcon"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M11 18h2v-2h-2zm1-16C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2m0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8m0-14c-2.21 0-4 1.79-4 4h2c0-1.1.9-2 2-2s2 .9 2 2c0 2-3 1.75-3 5h2c0-2.25 3-2.5 3-5 0-2.21-1.79-4-4-4"
    />
  </svg>
  <span
    className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
  />
</button>
`;

exports[`CustomIconButton should render correctly with QuestionMark icon by default 1`] = `
<button
  className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
  disabled={false}
  onBlur={[Function]}
  onContextMenu={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <svg
    aria-hidden={true}
    className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
    data-testid="QuestionMarkIcon"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M11.07 12.85c.77-1.39 2.25-2.21 3.11-3.44.91-1.29.4-3.7-2.18-3.7-1.69 0-2.52 1.28-2.87 2.34L6.54 6.96C7.25 4.83 9.18 3 11.99 3c2.35 0 3.96 1.07 4.78 2.41.7 1.15 1.11 3.3.03 4.9-1.2 1.77-2.35 2.31-2.97 3.45-.25.46-.35.76-.35 2.24h-2.89c-.01-.78-.13-2.05.48-3.15M14 20c0 1.1-.9 2-2 2s-2-.9-2-2 .9-2 2-2 2 .9 2 2"
    />
  </svg>
</button>
`;
